pyro/distributions/conditional_transformed_distribution.py

```diff
@@ -0,0 +1,157 @@
+import torch
+from torch.distributions import constraints
+from torch.distributions.distribution import Distribution
+from torch.distributions.transforms import Transform
+from torch.distributions.utils import _sum_rightmost
+from pyro.distributions import ConditionalTransform
+
+
+class ConditionalTransformedDistribution(Distribution):
+    r"""
+    Extension of the Distribution class, which applies a sequence of Transforms
+    to a base distribution.  Let f be the composition of transforms applied::
+
+        X ~ BaseDistribution
+        Y = f(X) ~ ConditionalTransformedDistribution(BaseDistribution, f)
+        log p(Y | Z=z) = log p(X | Z=z) + log |det (d(X | Z=z)/dY)|
+
+    Note that the ``.event_shape`` of a :class:`ConditionalTransformedDistribution` is the
+    maximum shape of its base distribution and its transforms, since transforms
+    can introduce correlations among events.
+
+    """
+    arg_constraints = {}
+
+    def __init__(self, base_distribution, transforms, validate_args=None):
+        self.base_dist = base_distribution
+        if isinstance(transforms, Transform):
```

Comment:
Why can't this be a `ConditionalTransform`?

Also this logic is gross, please simplify.

Reply:
Yes, you're right it can be a `ConditionalTransform` and the logic ought to be changed. I am hoping to eliminate having separate `Transform` and `ConditionalTransform` in a revised proposal so this issue will be eliminated